AccountTests/FxALoginStateMachineTests.swift

```diff
@@ -130,6 +143,20 @@ class FxALoginStateMachineTests: XCTestCase {
         self.waitForExpectationsWithTimeout(10, handler: nil)
     }
 
+    func testAdvanceFromEngagedAfterVerifiedWithoutNetwork() {
+        // Advancing from engaged after verified, but during outage, stays put.
+        withMachine(MockFxALoginClientWithoutNetwork()) { stateMachine in
+            let engagedState = FxAStateTests.stateForLabel(.EngagedAfterVerified)
+
+            let e = self.expectationWithDescription("Wait for login state machine.")
+            stateMachine.advanceFromState(engagedState, now: 0).upon { newState in
+                XCTAssertEqual(newState.label.rawValue, engagedState.label.rawValue)
```

Comment:
Mmm.  I'm not a fan of such implicit threading assumptions, although I guess it tends to be clear why a thing broke.